In [4]:
import pandas as pd
import numpy as np
import math
df = pd.read_csv('./HouseholderAtRisk.csv')

<h2>Task 1</h2>
<h4>1. What is the proportion of householders at risk?</h4>

In [5]:
print(df['AtRisk'].value_counts())

High    30498
Low      9501
Name: AtRisk, dtype: int64


Of 39999 observations, we found
High Risk : 30498 (76.247%)
Low Risk: 9501 (23.753%)

<h4>2.Did you have to fix any data quality problems?</h4>
The following attributes had quality problems:
<h5>Race</h5>
Most values in this attribute were missing.
The fix was to drop this column.
<h5>Age</h5>
There were a few values less than 1 and there were 968 missing values. 
These values were imputed with the mean value of 38.66.
<h5>WorkClass</h5>
The values were prepended with a space. The space was removed .
There were 2240 records with invalid value of "?". There were 972 missing values. These values were imputed with the mode "Private".
<h5>NumYearsEducation</h5>
There were 972 missing values. These values were replaced with the mean value of 10.
<h5>MaritalStatus</h5>
The values in this attribute were prepended with a space. The space was removed.

There were 972 missing values. These were replaced with the mode "Married-civ-spouse".
<h5>Occupation</h5>
The values in this attribute were prepended with a space. The space was removed.

There were 2246 records with invalid value of "?" and there were 986 missing values. These values were imputed with the mode "Prof-specialty".
<h5>Relationship</h5>
The values in this attribute were prepended with a space. The space was removed.

The 972 missing values were imputed with ‘Husband’ which is the mode. 
<h5>CapitalLoss</h5>
The 972 missing values were imputed with the mode of 0. Considering the values being skewed to the far left, it makes sense to impute 0 to the missing values.
<h5>CapitalGain</h5>
The 972 missing values were imputed with the mode of 0.
<h5>CapitalAvg</h5>
The 972 missing values were imputed with the mode of 0.
<h5>NumWorkingHoursPerWeek</h5>
There were 972 missing values. These values were imputed with the mean value of 40.
<h5>Sex</h5>
There were 972 missing values. These values were imputed with the mode of 0.
<h5>Country</h5>
The values in this attribute were prepended with a space. The space was removed.

699 values were ‘?’ - These were imputed with the mode‘United-States’.
30 missing values were imputed with ‘United-States’
917 values were ‘USA’ - These were changed to ‘United-States’
9 values were ‘US’ - These were changed to ‘United-States’
20 values were ‘Hong’ - These were changed to ‘Hong Kong’
97 values were South - ???

<h4>3. Irrelevant and redundant variables</h4>

<h5>ID</h5>
This attribute is a unique identifier and does not provide useful information for predicting the target variable.
<h5>Gender</h5>
This attribute is identical to Sex attribute but with different name. Sex attribute was chosen over this because when there are only two possible values it is better to transform it to binary variable. 
<h5>Education</h5>
Education attribute and NumYearsEducation is essentially a one-to-one mapping except that Education attribute is ordinal but NumYearsEducation is numeric.

<h4>3. What distribution scheme did you use? What data partitioning allocation did you set?</h4>
I've used N-fold cross validation with stratified sampling. This is our dataset is skewed.